app/webhook_test.go

```diff
@@ -470,3 +473,164 @@ func TestSplitWebhookPost(t *testing.T) {
+			}))
+			defer ts.Close()
+
+			channel := th.CreateChannel(th.BasicTeam)
+			hook, _ := createOutgoingWebhook(channel, ts.URL, th)
+			payload := getPayload(hook, th, channel)
+
+			th.App.TriggerWebhook(payload, hook, th.BasicPost, channel)
+
+			waitUntilWebhookResposeIsCreatedAsPost(channel, th, t, createdPost)
+
+			select {
+			case webhookPost := <-createdPost:
+				assert.Equal(t, webhookPost.Message, "sample response text from test server")
+				assert.Equal(t, webhookPost.Props["from_webhook"], "true")
+				if testCase.ExpectedIconUrl != "" {
+					assert.Equal(t, webhookPost.Props["override_icon_url"], testCase.ExpectedIconUrl)
+				} else {
+					assert.Nil(t, webhookPost.Props["override_icon_url"])
+				}
+
+				if testCase.ExpectedUsername != "" {
+					assert.Equal(t, webhookPost.Props["override_username"], testCase.ExpectedUsername)
+				} else {
+					assert.Nil(t, webhookPost.Props["override_username"])
+				}
+			case <-time.After(5 * time.Second):
+				t.Fatal("Timeout, webhook response not created as post")
+			}
+
+		})
+	}
+}
+
+func waitUntilWebhookResposeIsCreatedAsPost(channel *model.Channel, th *TestHelper, t *testing.T, createdPost chan *model.Post) {
+	go func() {
+		for i := 0; i < 5; i++ {
+			time.Sleep(time.Second)
+			posts, _ := th.App.GetPosts(channel.Id, 0, 5)
+			if len(posts.Posts) > 0 {
+				for _, post := range posts.Posts {
+					createdPost <- post
+					return
+				}
+			}
+		}
+	}()
+}
+func getPayload(hook *model.OutgoingWebhook, th *TestHelper, channel *model.Channel) *model.OutgoingWebhookPayload {
+	return &model.OutgoingWebhookPayload{
+		Token:       hook.Token,
+		TeamId:      hook.TeamId,
+		TeamDomain:  th.BasicTeam.Name,
+		ChannelId:   channel.Id,
+		ChannelName: channel.Name,
+		Timestamp:   th.BasicPost.CreateAt,
+		UserId:      th.BasicPost.UserId,
+		UserName:    th.BasicUser.Username,
+		PostId:      th.BasicPost.Id,
+		Text:        th.BasicPost.Message,
+		TriggerWord: "Abracadabra",
+		FileIds:     strings.Join(th.BasicPost.FileIds, ","),
+	}
+}
+
+func createOutgoingWebhook(channel *model.Channel, testCallBackUrl string, th *TestHelper) (*model.OutgoingWebhook, *model.AppError) {
+	outgoingWebhook := model.OutgoingWebhook{
+		ChannelId:    channel.Id,
+		TeamId:       channel.TeamId,
+		CallbackURLs: []string{testCallBackUrl},
+		Username:     "some-user-name",
+		IconURL:      "http://some-icon/",
+		DisplayName:  "some-display-name",
+		Description:  "some-description",
+		CreatorId:    th.BasicUser.Id,
+		TriggerWords: []string{"Abracadabra"},
+		ContentType:  "application/json",
+	}
+
+	return th.App.CreateOutgoingWebhook(&outgoingWebhook)
+}
+
+type TestCaseOutgoing struct {
+	EnablePostUsernameOverride bool
+	EnablePostIconOverride     bool
+	ExpectedUsername           string
+	ExpectedIconUrl            string
+}
+
+func getTestCases() map[string]TestCaseOutgoing {
+
+	testCasesOutgoing := make(map[string]TestCaseOutgoing)
```

Comment:
nit: I think it would be more idiomatic to write this as:
```go
testCasesOutgoing := map[string]TestCaseOutgoing{
    {
        ...
    },
    {
        ...
    }
}
```